src/01_getting_started/02_why_async.md

```diff
@@ -1,45 +1,121 @@
 # Why Async?
 
-We all love how Rust allows us to write fast, safe software. But why write
-asynchronous code?
+We all love how Rust allows us to write fast, safe software.
+But how does asynchronous programming fit into this vision?
 
-Asynchronous code allows us to run multiple tasks concurrently on the same OS
-thread. In a typical threaded application, if you wanted to download two
-different webpages at the same time, you would spread the work across two
-different threads, like this:
+Asynchronous programming, or async for short, is a _concurrent programming model_
+supported by an increasing number of programming languages.
+It lets you run a large number of concurrent
+tasks on a small number of OS threads, while preserving much of the
+look and feel of ordinary synchronous programming, through the
+`async/await` syntax.
+
+## Async vs other concurrency models
+
+Since the inception of concurrent programming, several models have been
```

Comment:
Nit: this could be less verbose; e.g. "There are multiple models of concurrent programming. Although details vary between languages and frameworks, it's worth understanding how asynchronous programming compares to other models."